db/hints/internal/hint_storage.hh

```diff
@@ -0,0 +1,31 @@
+/*
+ * Modified by ScyllaDB
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+#pragma once
+
+// Scylla includes.
+#include "db/commitlog/commitlog.hh"
+#include "db/commitlog/commitlog_entry.hh"
+#include "db/hints/internal/common.hh"
+#include "utils/loading_shared_values.hh"
+
+/// This file is supposed to gather meta information about data structures
+/// and types related to storing hints.
+///
+/// Under the hood, commitlog is used for managing, storing, and reading
+/// hints from disk.
+
+namespace db::hints {
+namespace internal {
+
+using node_to_hint_store_factory_type = utils::loading_shared_values<endpoint_id, db::commitlog>;
+using hints_store_ptr = node_to_hint_store_factory_type::entry_ptr;
+using hint_entry_reader = commitlog_entry_reader;
```

Comment:
Yes. An instance of `node_to_hint_store_factory_type` is part of shard hint manager, and the other two are used in `hint_host_manager.cc` and `hint_sender.cc`. I moved them here because:
* they're related to commitlog,
* `hints_store_ptr` depends on `node_to_hint_store_factory_type`, but they're used in two different files (`manager.hh` vs. `hint_host_manager.cc`).

I think this is a good place to place them.

They should become internal later on when commitlog has finally been isolated. It seems the alises for them in `manager.hh` are unnecessary at the moment, though. I'll remove them.

Reply:
Sorry, I got totally confused about the first two and didn't see the dependency between them. As for the last one (`hint_entry_reader`), we will see how the final version of the code looks like. Now, the hint sender is also responsible for the storage because it reads from the commitlog (using the aforementioned hint entry reader). Hopefully, it will make more sense after further refactors.